lib/grinder_files.dart

```diff
@@ -98,12 +100,210 @@ class FileSet {
+
+  /**
+   * Assume the current file system entity is a [File] and return it as such.
+   * You would call this instead of [entity] when the file system entity does
+   * not yet exist.
+   */
+  File get asFile => new File(path);
+
+  /**
+   * Assume the current file system entity is a [Directory] and return it as such.
+   * You would call this instead of [entity] when the file system entity does
+   * not yet exist.
+   */
+  Directory get asDirectory => new Directory(path);
+
+  /**
+   * Assume the current file system entity is a [Link] and return it as such.
+   * You would call this instead of [entity] when the file system entity does
+   * not yet exist.
+   */
+  Link get asLink => new Link(path);
+
+  /**
+   * Copy the the entity to the given destination. Return the newly created
+   * [Path].
+   */
+  Path copy(Path destDir) {
+    _copyImpl(entity, destDir.asDirectory);
+    return new Path(destDir).join(name);
+  }
+
+  /**
+   * Delete the entity at the path.
+   */
+  void delete() => _deleteImpl(entity);
+
+  /**
+   * Synchronously create the file. See also [File.create].
+   *
+   * If [recursive] is false, the default, the file is created
+   * only if all directories in the path exist. If [recursive] is true, all
+   * non-existing path components are created.
+   */
+  void createFile({bool recursive: false}) {
+    asFile.createSync(recursive: recursive);
+  }
+
+  /**
+   * Synchronously create the directory. See also [Directory.create].
+   *
+   * If [recursive] is false, the default, the file is created
+   * only if all directories in the path exist. If [recursive] is true, all
+   * non-existing path components are created.
+   */
+  void createDirectory({bool recursive: false}) {
+    asDirectory.createSync(recursive: recursive);
+  }
+
+  /**
+   * Synchronously create the link. See also [Link.create].
+   *
+   * If [recursive] is false, the default, the file is created
+   * only if all directories in the path exist. If [recursive] is true, all
+   * non-existing path components are created.
+   */
+  void createLink(String target, {bool recursive: false}) {
+    asLink.createSync(target, recursive: recursive);
+  }
+
+  /**
+   * Join the given path elements to this path, and return a new [Path] object.
+   */
+  Path join([arg0, String arg1, String arg2, String arg3, String arg4,
+    String arg5, String arg6, String arg7, String arg8, String arg9]) {
+    List args = [];
+
+    if (arg0 is List) {
+      args = arg0;
+    } else if (arg0 is String) {
+      _addNonNull(args, arg0);
+      _addNonNull(args, arg1);
+      _addNonNull(args, arg2);
+      _addNonNull(args, arg3);
+      _addNonNull(args, arg4);
+      _addNonNull(args, arg5);
+      _addNonNull(args, arg6);
+      _addNonNull(args, arg7);
+      _addNonNull(args, arg8);
+      _addNonNull(args, arg9);
+    }
+
+    if (args.isEmpty) {
+      return this;
+    } else {
+      return new Path('${path}${_sep}${args.join(_sep)}');
```

Comment:
If you do `var paths = [path];` above, instead of `var args = [];`, then this can change to just `new Path(paths.join(_sep))`.